docs/changes/io.fits/14576.feature.rst

```diff
@@ -0,0 +1,2 @@
+Do not call ``gc.collect()`` in ``CompImageHDU.compressed_data`` as it has
```

Comment:
Specifically in the deleter right? Maybe rephrase to say when closing a ConpImageHDU since that is when people are most likely to call this indirectly?